### PR TITLE
feat: decouple account selector from qr code connector

### PR DESCRIPTION
**Pull request by stanleyyconsensys**
## **Description**

In order to enable multiple account select in ledger hardware wallet, 
This PR is focus on decoupling the `Account selector component` from QR hardware wallet for reusable purpose

The changes should not impact to any UI / operations differences compare to existing QR hardware wallet account selector

For clean code purpose, the changes also decoupling the method `toBlockExplorer` from account selector component into a new hook `useBlockExplorer`

**Logical flow**
image

## **Related issues**

issue : [109](https://app.zenhub.com/workspaces/cet-metamask-hardware-wallets-6566ee6aedc46007d5a260bb/issues/zh/109)

## **Manual testing steps**

1. Go to main screen
2. Tab current account to open account drawer
3. Click add account or hardware account
4. Choose add hardware account
5. Choose QR based
6. Scan Qr wallet
7. Account selector should appear after scan success
8. Account selector should able to perform select/un select account, unlock accounts, next page, prev pages operation as what it has to be in previous version

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/UI/HardwareWallet/AccountSelector/hooks.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    });
+    if (unTrackedAccounts.length > 0) {
+      AccountTrackerController.syncBalanceWithAddresses(unTrackedAccounts).then(
+        (_trackedAccounts: AccountBalances) => {
+          setTrackedAccounts(
+            Object.assign({}, trackedAccounts, _trackedAccounts),
+          );
+        },
+      );
+    }
```
Comment (Akaryatrh) [inline app/components/UI/HardwareWallet/AccountSelector/hooks.tsx:38]: Out of curiosity, there's no error handling needed here? If not, maybe you could switch with a simple await?
```suggestion
      const _trackedAccounts = await AccountTrackerController.syncBalanceWithAddresses(unTrackedAccounts)
      setTrackedAccounts({...trackedAccounts, ..._trackedAccounts})
```

**Review by Akaryatrh — DISMISSED** [review]

**Review by Akaryatrh — DISMISSED** [review]

**Review by Akaryatrh — DISMISSED** [review]

**Comment by vivek-consensys** [comment]
E2E Manual Test:- 


https://github.com/MetaMask/metamask-mobile/assets/106310394/b16f0b40-695e-4994-ab75-38ffe634f931

**Comment by gantunesr** [comment]
I found a minor bug unrelated to this PR, it'd be good to open a ticket for it cc @angelcheung22 

https://github.com/MetaMask/metamask-mobile/assets/17601467/53d0b3ee-ebd3-41f8-9d24-5119c1a65f48

**Review by gantunesr — COMMENTED** [review]
The rest of the code looks good to me

**Review by dawnseeker8 — DISMISSED** [review]

**Comment by angelcheung22** [comment]
> I found a minor bug unrelated to this PR, it'd be good to open a ticket for it cc @angelcheung22
> 
>  RPReplay_Final1705930225.mp4

Hey Gustavo, im confused about the bug that you are showing in the video. May I know whats the expected steps?

File: app/components/UI/HardwareWallet/AccountSelector/hooks.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+export interface AccountBalance {
+  balance: string;
+}
+
+export interface AccountBalances {
+  [p: string]: AccountBalance;
+}
+
+export const useAccountsBalance = (accounts: IAccount[]) => {
```
[the remark this answers is not in the document]
Comment (dawnseeker8) [reply app/components/UI/HardwareWallet/AccountSelector/hooks.tsx:16]: have added a unit tests for hooks.tsx

File: app/components/hooks/useBlockExplorer.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import { findBlockExplorerForRpc } from '../../util/networks';
+import { getEtherscanAddressUrl } from '../../util/etherscan';
+import { useSelector } from 'react-redux';
+import { useNavigation } from '@react-navigation/native';
+import {
+  selectNetworkConfigurations,
+  selectProviderConfig,
+} from '../../selectors/networkController';
+
+const useBlockExplorer = () => {
```
[the remark this answers is not in the document]
Comment (dawnseeker8) [reply app/components/hooks/useBlockExplorer.ts:13]: have added a unit tests for hooks.tsx

**Comment by dawnseeker8** [comment]
i have implemented two unit tests for `useBlockExplorer` and `hooks.tsx` under `hardwareWallet/AccountSelector/hooks.tsx` based on @gantunesr comments in code review.
https://github.com/MetaMask/metamask-mobile/pull/8093

now This PR is ready to review again.

**Review by dawnseeker8 — APPROVED** [review]

**Comment by dawnseeker8** [comment]
Regarding the small bug @gantunesr report, i have tried with `main` branch and confirm that this is pre-existing bug in mobile. every etherscan link open in QR code import screen, if you click `back` in that screen, UI will go back to main screen.

**Comment by angelcheung22** [comment]
> Regarding the small bug @gantunesr report, i have tried with `main` branch and confirm that this is pre-existing bug in mobile. every etherscan link open in QR code import screen, if you click `back` in that screen, UI will go back to main screen.

https://app.zenhub.com/workspaces/cet-metamask-hardware-wallets-6566ee6aedc46007d5a260bb/issues/zh/139 <-per discussion from last weekly with alex, this one can be low priority and add to backlog for future fixing. Added new ticket in zenhub

**Review by Akaryatrh — APPROVED** [review]

**Review by gantunesr — APPROVED** [review]

**Pull request merged**